src/lib/metrics_extractor/base_metrics_extractor.py

```diff
@@ -72,12 +71,15 @@ def get_last_update_time(self, timestream_query_client) -> datetime:
         return last_date
 
     @abstractmethod
-    def prepare_metrics_data(self, since_time: datetime) -> (list, dict):
+    def prepare_metrics_data(
+        self, since_time: datetime, result_ids: list = []
```

Comment:
I'd consider another way of implementing an additional parameter, which is specific to Glue DQ.

1. don't add a parameter to this method (so you won't need to change all other classes implementation)
2. in glue DQ metrics extractor add set_results_id  (self.results_ids = results_ids)
3. in glue DQ's prepare_metrics_data implementation refer to self.results_id
4. in lambda_extract_metric add something like 
if Glue DQ:
   extractor.set_result_ids
   
   Hopefully, this makes sense. Anyway, it'd be nice avoid changing all other classes implementation just for one-class specific parameter

Reply:
sounds good to me, thanks a lot!